### PR TITLE
Obfuscate JWT token value while logging requests in ingestion

### PR DESCRIPTION
**Pull request by polyakrecords**
Things to consider before you submit the PR:

* [ ] Has `CHANGELOG.md` been updated?
* [X] Are tests passing locally?
* [X] Are the files formatted correctly?
* [X] Did you add unit tests?
* [ ] Did you check UI tests on the sample app? They are not executed on CI.
* [ ] Did you test your change with either the sample apps that are included in the repository or with a blank app that uses your change?

## Description

Changed obfuscation method for JWT authorization header value in ingestion logs.

### CODE REVIEW

File: AppCenter/AppCenter/Internals/Ingestion/MSAppCenterIngestion.m [hunk 1]
```diff
@@ -97,9 +97,12 @@ - (NSURLRequest *)createRequest:(NSObject *)data eTag:(NSString *)__unused eTag
 }
 
 - (NSString *)obfuscateHeaderValue:(NSString *)value forKey:(NSString *)key {
-  return ([key isEqualToString:kMSHeaderAppSecretKey] || [key isEqualToString:kMSAuthorizationHeaderKey])
-             ? [MSIngestionUtil hideSecret:value]
-             : value;
+  if ([key isEqualToString:kMSAuthorizationHeaderKey]) {
+    return [NSString stringWithFormat:kMSBearerTokenHeaderFormat, @"***"];
```
Comment (jaelim-ms) [inline AppCenter/AppCenter/Internals/Ingestion/MSAppCenterIngestion.m]: Add a util method in `MSIngestionUtil` and handle the value similar with `hideSecret`

Comment (jaelim-ms) [reply AppCenter/AppCenter/Internals/Ingestion/MSAppCenterIngestion.m]: There is a good example implemented in this [PR](https://github.com/Microsoft/AppCenter-SDK-Apple/pull/1353/files#diff-a606764e07115ebee32c50532b58e867R44)
We probably move this to base class.

File: AppCenter/AppCenter/Internals/Ingestion/MSAppCenterIngestion.m [hunk 2]
```diff
@@ -97,9 +97,12 @@ - (NSURLRequest *)createRequest:(NSObject *)data eTag:(NSString *)__unused eTag
 }
 
 - (NSString *)obfuscateHeaderValue:(NSString *)value forKey:(NSString *)key {
-  return ([key isEqualToString:kMSHeaderAppSecretKey] || [key isEqualToString:kMSAuthorizationHeaderKey])
-             ? [MSIngestionUtil hideSecret:value]
-             : value;
+  if ([key isEqualToString:kMSAuthorizationHeaderKey]) {
+    return [NSString stringWithFormat:kMSBearerTokenHeaderFormat, [MSIngestionUtil hideAuthToken:value]];
+  }  else if ([key isEqualToString:kMSHeaderAppSecretKey]) {
```
Comment (MatkovIvan) [inline AppCenter/AppCenter/Internals/Ingestion/MSAppCenterIngestion.m]:
```suggestion
  } else if ([key isEqualToString:kMSHeaderAppSecretKey]) {
```

File: AppCenter/AppCenter/Internals/Ingestion/MSAppCenterIngestion.m [hunk 3]
```diff
@@ -97,9 +97,12 @@ - (NSURLRequest *)createRequest:(NSObject *)data eTag:(NSString *)__unused eTag
 }
 
 - (NSString *)obfuscateHeaderValue:(NSString *)value forKey:(NSString *)key {
-  return ([key isEqualToString:kMSHeaderAppSecretKey] || [key isEqualToString:kMSAuthorizationHeaderKey])
-             ? [MSIngestionUtil hideSecret:value]
-             : value;
+  if ([key isEqualToString:kMSAuthorizationHeaderKey]) {
+    return [NSString stringWithFormat:kMSBearerTokenHeaderFormat, [MSIngestionUtil hideAuthToken:value]];
```
Comment (MatkovIvan) [inline AppCenter/AppCenter/Internals/Ingestion/MSAppCenterIngestion.m]: Move all logic to `MSIngestionUtil`

File: AppCenter/AppCenter/Internals/Ingestion/Util/MSIngestionUtil.m [hunk 4]
```diff
@@ -35,4 +36,11 @@ + (NSString *)hideSecret:(NSString *)secret {
   return [secret stringByReplacingCharactersInRange:NSMakeRange(0, appSecretHiddenPart.length) withString:appSecretHiddenPart];
 }
 
++ (NSString *)hideAuthToken:(NSString *)token {
+
+  // Hide token value
```
Comment (MatkovIvan) [inline AppCenter/AppCenter/Internals/Ingestion/Util/MSIngestionUtil.m]:
```suggestion
  // Hide token value.
```

**Pull request merged**